arbitrator/src/arbitrator_node.cpp

```diff
@@ -100,7 +100,7 @@ namespace arbitrator
                                 [this]() {this->arbitrator_->bumper_pose_cb();});
         
         arbitrator_run_ = create_timer(get_clock(),
-                                std::chrono::duration<double>(1/config_.planning_frequency),
+                                std::chrono::duration<double>(1/(config_.planning_frequency * 2 )), //there is waiting state between each planning state
```

Comment:
The intention is to keep the arbitrator planning at intended frequency - 1s. Looks like ROS2 conversion kept the WAITING state from ROS1, which occurs between PLANNING state in the state machine. So if the timer calls state callback each 1 second, the arbitrator ends up planning every 2s due to PLANNING > WAITING > PLANNING transition. That's why the frequency is doubled.

Reply:
Ok @aswath1, I think what you just said should be it's own comment in the code. The bit you added is helpful but it could be clearer if you copy and paste your paragraph in.